source/docs/software/wpilib-tools/shuffleboard/custom-widgets/creating-custom-data-types.rst

```diff
@@ -0,0 +1,151 @@
+Creating Custom Data Types
+==========================
+
+Widgets allow us to control and visualize different types of data. This data can be numerical in the form to integers, double, longs, etc..., or less primitive forms of data, such as Java objects, or data involving multiple fields. In order to display these types of data using widgets, it is helpful to create a container class for them.
```

Comment:
`This data can be numerical in the form to integers, double, longs, etc..., or less primitive forms of data, such as Java objects, or data involving multiple fields.` This is awkwardly worded